app/models/host/managed.rb

```diff
@@ -880,16 +880,23 @@ def build_status_label(options = {})
   # rebuilds orchestration configuration for a host
   # takes all the methods from Orchestration modules that are registered for configuration rebuild
   # returns  : Hash with 'true' if rebuild was a success for a given key (Example: {"TFTP" => true, "DNS" => false})
-  def recreate_config
+  def recreate_config(only = nil)
     result = {}
-    Nic::Managed.rebuild_methods.map do |method, pretty_name|
+    nic_rebuild_methods = Nic::Managed.rebuild_methods
+    host_rebuild_methods = self.class.rebuild_methods
```

Comment:
@lzap Should the error checking be done in Host::Managed method or should this be done in API where the inputs are supplied?